lib/codecs/src/decoding/mod.rs

```diff
@@ -216,26 +227,45 @@ impl tokio_util::codec::Decoder for Framer {
     }
 }
 
-/// Configuration for building a `Deserializer`.
+/// Deserializer configuration.
 // Unfortunately, copying options of the nested enum variants is necessary
 // since `serde` doesn't allow `flatten`ing these:
 // https://github.com/serde-rs/serde/issues/1402.
 #[configurable_component]
 #[derive(Clone, Debug)]
 #[serde(tag = "codec", rename_all = "snake_case")]
+#[configurable(description = "Configures how events are decoded from raw bytes.")]
+#[configurable(metadata(docs::enum_tag_description = "The codec to use for decoding events."))]
 pub enum DeserializerConfig {
-    /// Configures the `BytesDeserializer`.
+    /// Uses the raw bytes as-is.
     Bytes,
-    /// Configures the `JsonDeserializer`.
+
+    /// Decodes the raw bytes as [JSON][json].
+    ///
+    /// [json]: https://en.wikipedia.org/wiki/JSON
```

Comment:
Thinking out loud, we should probably standardize/have guidelines on what to link. Spec site vs wiki vs whatever..

Reply:
Yeah, that's a fair point. I'll go and find the most official link I reasonably can for these and update this stuff.

Reply:
What do you mean by the "variation" in "targets"? Like the quality/official-ness of a given link?